### PR TITLE
Blog post for 1.1.0 release

### PR DESCRIPTION
**Pull request by benasher44**
CocoaPods 1.1.0 release blog post. Improvements mentioned:
- Extension and framework improvements
- Xcode 8 support
- Error handling improvements

More?

### CODE REVIEW

**Comment by benasher44** [comment]
I haven't been able to run `rake serve:drafts` successfully yet. I hit this error:
```
blog.cocoapods.org [basher_1.1.0] $ rake serve:drafts --trace
** Invoke serve:drafts (first_time)
** Execute serve:drafts
Starting the server locally on http://localhost:4000
bundle exec jekyll serve --watch --drafts --port 4000
Configuration file: /Users/benasher/Code/Ruby/blog.cocoapods.org/_config.yml
            Source: /Users/benasher/Code/Ruby/blog.cocoapods.org
       Destination: /Users/benasher/Code/Ruby/blog.cocoapods.org/_site
      Generating... 
  Liquid Exception: undefined method `[]' for nil:NilClass in _layouts/post.html
jekyll 2.5.3 | Error:  undefined method `[]' for nil:NilClass
rake aborted!
Command failed with status (1): [bundle exec jekyll serve --watch --drafts ...]
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/file_utils.rb:66:in `block in create_shell_runner'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/file_utils.rb:57:in `call'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/file_utils.rb:57:in `sh'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/file_utils_ext.rb:37:in `sh'
/Users/benasher/Code/Ruby/blog.cocoapods.org/Rakefile:28:in `block (2 levels) in <top (required)>'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/Users/benasher/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/bin/rake:23:in `load'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/bin/rake:23:in `<main>'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/bin/ruby_executable_hooks:15:in `eval'
/Users/benasher/.rvm/gems/ruby-2.0.0-p648/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => serve:drafts
```
I don't have time now, but I do plan on debugging this later.

**Comment by benasher44** [comment]
@DanToml a few questions for you:

1. I feel like I'm missing some ways in which you can specify the Swift version in the podspec. I went based on your PR, but it seems like there might be pre-existing ways: `spec.pod_target_xcconfig` perhaps?.
2. I didn't mention the `Dir.chdir` issue, but that a lot of what was going on there went over my head. This is mostly because I haven't spent much time digesting those parts of Xcodeproj. Do you think this deserves mentioning (cc @segiddins)? This was quite a wrench thrown at us by Xcode 8.

**Comment by orta** [comment]
So, believe it or not 1.1 contains a pretty useful feature from me too, big enough to warrant a mention.


> ### Error Handling Improvements

> Now when CocoaPods receives unexpected errors it will highlight existing issues that might be related in the console. This makes it easier to find the answer to common issues. We do this by using [gh_inspector](https://github.com/orta/gh_inspector) which you may have seen [inside Fastlane](https://github.com/fastlane/fastlane/releases/tag/1.96.0) too.

**Comment by benasher44** [comment]
Definitely! Sorry I missed It. I think I've just had Xcode 8 / extension tunnel vision the past few months. Will update the PR soon.

If anyone has any other big updates I missed, please chime in!

**Comment by benasher44** [comment]
@orta added verbatim + a comma.

**Comment by DanToml** [comment]
@benasher44 

1) the pod target xcconfig will mean people can't integrate with the wrong version I think, which is nice for correctness. .swift-version is solely for linting/pushing.

2) I think it's probably worth mentioning, because we sank a lot of time into it. Honestly still not sure of the underlying cause, but it looks like DevToolsCore might be triggering a kernel bug?

**Comment by benasher44** [comment]
pod target xcconfig won't do both?

For the DevToolsCore issue, I'm happy to include a section if you write it up 😀. It could just be a sentence or two in the Xcode 8 section

**Comment by DanToml** [comment]
@benasher44 I don't think so, because it needs to be in the root project not just the pod targets?

File: _drafts/1.1.0.markdown [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+In every case, `pod install` looks through each of your project's non-framework target's "Target Dependencies" build phase to find where the framework is meant to be integrated (same for extensions as well). For framework-only projects, if CocoaPods can tell that your project only contains framework targets, then it assumes that the project must be a framework-only project (for doing framework development) and emits a warning about this assumption. In that case, CocoaPods doesn't have much to do except ensure your project's targets have the headers to build properly.
+
+
+### Xcode 8 and iOS 10
+
+Xcode 8 is here! It brings support for iOS 10, macOS Sierra, new versions of Swift, and new challenges for CocoaPods. The first is the ability to support Swift 2.3 and 3.0 projects and communicate this in a podspec. In CocoaPods 1.1.0, setting the version of Swift that your pod uses is supported in following ways:
+
+* It defaults to Swift 2.3, if your pod uses Swift.
+* It uses setting from your `.swift-version` file if present (inspired by [swiftenv](https://github.com/kylef/swiftenv).
```
Comment (orta) [inline _drafts/1.1.0.markdown]: We should include a reference of what this looks like. 

> * It uses setting from your `.swift-version` file if present (inspired by [swiftenv](https://github.com/kylef/swiftenv).
>   This is a file that only contains the swift version as text: e.g. for Swift 3, you would have `3.0` in the file.

File: _drafts/1.1.0.markdown [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title:  "CocoaPods 1.1.0 - Improved support for extensions and frameworks, Xcode 8 support, and error handling improvements"
+author: nickname
+categories: cocoapods releases
+---
+TL;DR: _CocoaPods 1.1.0_ has been released, and it comes with improved support for extensions and frameworks, Xcode 8 support, and error handling improvements.
```
Comment (segiddins) [inline _drafts/1.1.0.markdown]: might want a `<!-- more -->` somewhere in here

**Comment by benasher44** [comment]
@DanToml if the project is swift version unspecified, then I *think* all that matters is all of the swift pods being integrated need to have the compatible swift versions

**Comment by benasher44** [comment]
Oh wait no that's clearly wrong because of CocoaPods/CocoaPods#5846. Hm. Maybe we should just be using the pod's build settings a source of truth for setting the swift version for validation. Then, `.swift-version` file just becomes another way to set the pod's swift version setting. I think this might help some of the existing swift version confusion for pushing pods.

**Comment by benasher44** [comment]
Also, I'm waiting on the resolution of CocoaPods/CocoaPods#5860. It appears I may have gotten it wrong for standalone messages applications. If that's that the case, then messages extensions are just like any other extension, and I should remove that section from the post.

**Comment by benasher44** [comment]
The other possibility for this swift version issue is the swift version unspecified is different than the swift version setting is not set at all.

**Comment by AliSoftware** [comment]
It doesn't feel clear to me how pod authors are supposed to set the Swift version if they support multiple.

For example I have pods that support both 2.2, 2.3 and 3.0 in the same master branch. Should I create a `.swift-version` file containing `3.0` so that if the hosting project is itself 3.0 then CP will set the Swift version of the pod target to 3.0 and profit of the more modern API of my pod? Or should I not create that `.swift-version` file so that CP knows it's compatible with both 2.3 and 3.0? And in that case will it default to 2.3 even if the hosting user project is set to use 3.0?!

**Comment by orta** [comment]
It only sets the Swift version inside the linter, if your lib supports source-compatible 2.3 and 3.0 then 2.3 is fine. We keep no metadata around.

File: _drafts/1.1.0.markdown [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+* Framework(s) in your application-type project
+* Framework(s) in your application-type project, where the framework may live in a sub-project (i.e. an Xcode project within your Xcode project)
+* Framework(s)-only projects (for doing framework development)
+
+In every case, `pod install` looks through each of your project's non-framework target's "Target Dependencies" build phase to find where the framework is meant to be integrated (same for extensions as well). For framework-only projects, if CocoaPods can tell that your project only contains framework targets, then it assumes that the project must be a framework-only project (for doing framework development) and emits a warning about this assumption. In that case, CocoaPods doesn't have much to do except ensure your project's targets have the headers to build properly.
+
+
+### Xcode 8 and iOS 10
+
+Xcode 8 is here! It brings support for iOS 10, macOS Sierra, new versions of Swift, and new challenges for CocoaPods. The first is the ability to support Swift 2.3 and 3.0 projects and communicate this in a podspec. In CocoaPods 1.1.0, setting the version of Swift that your pod uses is supported in following ways:
```
Comment (AliSoftware) [inline _drafts/1.1.0.markdown]: > […] setting the Swift version your pod is gonna be **verified and linted against** can be done is the following ways:
> …
> 
> Note that this only affect the linting. During integration of a pod, it's Swift version is set to the same version as the one used in the user's hosting project.

**Comment by AliSoftware** [comment]
@orta got it. Then might need some more explicit explanations fort that paragraph then (suggestion above)

File: _drafts/1.1.0.markdown (hunk 3 above)
Comment (benasher44) [inline _drafts/1.1.0.markdown]: I'm going to rework this section soon. There are still a few Swift-related PRs to come I think.  I'm going to wait for Swift support to stabilize before I update.

File: _drafts/1.1.0.markdown [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+* It defaults to Swift 2.3, if your pod uses Swift.
+* It uses setting from your `.swift-version` file if present (inspired by [swiftenv](https://github.com/kylef/swiftenv). This is a file that only contains the swift version as text: e.g. for Swift 3, you would have 3.0 in the file.
```
Comment (orta) [inline _drafts/1.1.0.markdown]: RC3 added `--swift-version=VERSION` - https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md#110rc3-2016-10-11

**Comment by benasher44** [comment]
In case anyone hits that backtrace again, it was because I hadn't added myself to _config.yml and configured the author properly (should have fully read the README).

**Comment by benasher44** [comment]
Alright, this is much closer to being ready now. I think the only thing left to add is something about CocoaPods/cocoadocs.org#473 if we do something there.

File: _drafts/1.1.0.markdown [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+* It [defaults to Swift 3.0](https://github.com/CocoaPods/CocoaPods/issues/6041), if your pod uses Swift.
+* It uses setting from your `.swift-version` file if present (inspired by [swiftenv](https://github.com/kylef/swiftenv).
+* The `--swift-version` flag can be used to specify a version of Swift to use. This overrides the version set in the `.swift-version` file.
+
+Next, iOS 10 makes some new extension and application types available. In particular, CocoaPods has been updated to recognize the messages extensions and application product types.
+
+
+### About Supporting Multiple Swift Versions in the Podspec
+
+Supporting the Swift language version used by a pod as part of the podspec is tricky. There are a few different options we considered, but we ultimately decided not to add any new podspec dsl for this release. Here are some examples of options we considered for this release:
```
Comment (alloy) [inline _drafts/1.1.0.markdown]: I’m missing details on _why_ it’s so tricky to deal with Swift versions at all. E.g. Swift doesn’t use semantic versioning?

**Comment by alloy** [comment]
I’m missing updates on all of the speed improvements done in this release, I believe there were great optimisations done, no?

**Comment by orta** [comment]
Speed improvements may have been caused by the previous 1.1.0 releases - not sure personally

**Comment by DanToml** [comment]
@alloy @orta 
Performance improvements were mostly in https://github.com/CocoaPods/CocoaPods/pull/5927, https://github.com/CocoaPods/CocoaPods/pull/5934, and https://github.com/CocoaPods/CocoaPods/pull/5837.

@dnkoutso can probably elaborate more.

**Comment by dnkoutso** [comment]
The performance improvements were mostly done for larger projects during `pod install` by caching various methods that are expensive because of a lot of file access.

I think the tl;dr is that `pod install time is significantly improved, especially for large projects`.

File: _drafts/1.1.0.markdown (hunk 5 above)
Comment (benasher44) [inline _drafts/1.1.0.markdown]: 👍. Let me know if you think it's still unclear, and I'm happy to elaborate further!

File: _drafts/1.1.0.markdown [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+2. This could be helped by instead supporting a `min_swift_version` option to indicate the minimum Swift version the pod supports. This wouldn't force authors to update their pods when new versions come out, but it would break if a new version of Swift comes out that is not backwards compatible with the one used to build the pod.
+
+3. We could add a `swift_version_range` option to indicate the range of Swift versions the pod supports, but this suffers from the same issues as option 2. There is no way to verify that a pod pushed today that claims to support up through Swift version 4 actually supports that version.
+
+In this release of CocoaPods, you lint your pod against the version of Swift you intend to support. When integrating pods, CocoaPods generates pod targets that use the Swift version set in your project. This allows pod authors the flexibility to support multiple incompatible swift versions if they wish by using `#if swift()` checks, while keeping the pod maintenance burden light across Swift 3 versions.
+
```
Comment (benasher44) [inline _drafts/1.1.0.markdown]: Is the plan to update this with a link once 1.1.0 is live, and we have an example from people pushing their specs?

Comment (orta) [inline _drafts/1.1.0.markdown]: Yeah, I can do that 👍

**Pull request merged**